### PR TITLE
Redone set sync encryption for Android to match Chromium 89

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Redoing set sync encryption on Android to match Chromium changes

Resolves https://github.com/brave/brave-browser/issues/13754

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by samartnik — APPROVED** [review]
lgtm

**Review by darkdh — COMMENTED** [review]
It's better to address https://github.com/brave/brave-browser/issues/10454 sooner, otherwise following chromium upgrades might break sync encryption on Android in the future

File: browser/android/brave_sync_worker.cc [hunk 1]
```diff
@@ -239,117 +239,69 @@ void BraveSyncWorker::OnResetDone() {
...
-namespace {
+void BraveSyncWorker::SetEncryptionPassphrase() {
+  syncer::SyncService* service = GetSyncService();
+  DCHECK(service);
+  DCHECK(service->IsEngineInitialized());
+
+  syncer::SyncUserSettings* sync_user_settings =
+      GetSyncService()->GetUserSettings();
+
+  bool successfully_set = false;
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply browser/android/brave_sync_worker.cc]: thanks, fixed

File: browser/android/brave_sync_worker.cc [hunk 2]
```diff
@@ -239,117 +239,69 @@ void BraveSyncWorker::OnResetDone() {
...
+
+void BraveSyncWorker::SetDecryptionPassphrase() {
+  syncer::SyncService* service = GetSyncService();
+  DCHECK(service);
+  DCHECK(service->IsEngineInitialized());
+
+  syncer::SyncUserSettings* sync_user_settings =
+      GetSyncService()->GetUserSettings();
+
+  bool successfully_set = false;
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply browser/android/brave_sync_worker.cc]: thanks, fixed

File: browser/android/brave_sync_worker.cc [hunk 3]
```diff
@@ -239,117 +239,69 @@ void BraveSyncWorker::OnResetDone() {
...
-      }
-    }
-  }
-
-  if (passphrase_failed ||
-      service->GetUserSettings()->IsPassphraseRequiredForPreferredDataTypes()) {
-    VLOG(1) << __func__ << " setup passphrase failed";
-  }
+  SetEncryptionPassphrase();
+  SetDecryptionPassphrase();
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply browser/android/brave_sync_worker.cc]: Thanks fixed at https://github.com/brave/brave-core/pull/7697/files#diff-485d3253973f7504bec1b8e4cb51394ea4fdfa4cc34eef91057664f243a726fdR289

But previous version also had that checks inside both of `SetEncryptionPassphrase`/`SetDecryptionPassphrase` , but now this is more obvious.

**Comment by AlexeyBarabash** [comment]
> It's better to address [brave/brave-browser#10454](https://github.com/brave/brave-browser/issues/10454) sooner, otherwise following chromium upgrades might break sync encryption on Android in the future

I guess, yes, probably will do after resolving issues with yellow status of transient errors.
And from another side, java sync code can also be changed, so that will not be a remedy.

**Review by darkdh — APPROVED** [review]

**Pull request merged**